src/app/shared/issue-tables/issue-tables.component.ts

```diff
@@ -158,63 +154,69 @@ export class IssueTablesComponent implements OnInit, AfterViewInit {
+      duration: this.snackBarAutoCloseTime
+    });
+    snackBarRef.onAction().subscribe(() => {
+      this.undeleteIssue(id, event, false);
+    });
+  }
+
+  deleteIssue(id: number, event: Event, actionUndoable: boolean = true) {
     this.logger.info(`IssueTablesComponent: Deleting Issue ${id}`);
-    this.issuesPendingDeletion = {
-      ...this.issuesPendingDeletion,
-      [id]: true
-    };
+
+    this.issuesPendingDeletion = { ...this.issuesPendingDeletion, [id]: true };
     this.issueService
       .deleteIssue(id)
-      .pipe(
-        finalize(() => {
-          const { [id]: issueRemoved, ...theRest } = this.issuesPendingDeletion;
-          this.issuesPendingDeletion = theRest;
-        })
-      )
+      .pipe(finalize(() => this.handleIssueDeletionFinalization(id, event, actionUndoable)))
       .subscribe(
         (removedIssue) => {},
         (error) => {
           this.errorHandlingService.handleError(error);
         }
       );
     event.stopPropagation();
+  }
 
+  private handleIssueRestorationFinalization(id: number, event: Event, actionUndoable: boolean) {
+    const { [id]: issueRestored, ...theRest } = this.issuesPendingRestore;
+    this.issuesPendingRestore = theRest;
+
+    if (!actionUndoable) {
+      return;
+    }
     let snackBarRef = null;
     snackBarRef = this.snackBar.openFromComponent(UndoActionComponent, {
-      data: { message: `Deleted issue ${id}` },
+      data: { message: `Restored issue ${id}` },
       duration: this.snackBarAutoCloseTime
     });
     snackBarRef.onAction().subscribe(() => {
-      this.undeleteIssue(id, event);
+      this.deleteIssue(id, event, false);
     });
   }
 
-  undeleteIssue(id: number, event: Event) {
+  undeleteIssue(id: number, event: Event, actionUndoable: boolean = true) {
     this.logger.info(`IssueTablesComponent: Undeleting Issue ${id}`);
-    this.issueService.undeleteIssue(id).subscribe(
-      (reopenedIssue) => {},
-      (error) => {
-        this.errorHandlingService.handleError(error);
-      }
-    );
-    event.stopPropagation();
-
-    this.snackBar.open(`Restored issue ${id}`, '', { duration: this.snackBarAutoCloseTime });
-  }
-
-  openDeleteDialog(id: number, event: Event) {
-    const dialogRef = this.dialogService.openUserConfirmationModal(
-      this.deleteIssueModalMessages,
-      this.yesButtonModalMessage,
-      this.noButtonModalMessage
-    );
 
-    dialogRef.afterClosed().subscribe((res) => {
-      if (res) {
-        this.logger.info(`IssueTablesComponent: Deleting issue ${id}`);
-        this.deleteIssue(id, event);
-      }
-    });
+    this.issuesPendingRestore = { ...this.issuesPendingRestore, [id]: true };
+    this.issueService
+      .undeleteIssue(id)
+      .pipe(finalize(() => this.handleIssueRestorationFinalization(id, event, actionUndoable)))
+      .subscribe(
+        (restoredIssue) => {},
+        (error) => {
+          this.errorHandlingService.handleError(error);
+        }
```

Comment:
Same as above

This can be combined into:

```suggestion
      .subscribe(
        (restoredIssue) => this.handleIssueRestorationFinalization(id, event, actionUndoable),
        (error) => this.errorHandlingService.handleError(error)
      );
    event.stopPropagation();
```

Reply:
Same as above